lib/components/narrative/line-itin/connected-itinerary-body.js

```diff
@@ -0,0 +1,80 @@
+import isEqual from 'lodash.isequal'
+import TriMetLegIcon from '@opentripplanner/icons/lib/trimet-leg-icon'
+import TransitLegSummary from '@opentripplanner/itinerary-body/lib/defaults/transit-leg-summary'
+import ItineraryBody from '@opentripplanner/itinerary-body/lib/otp-react-redux/itinerary-body'
+import LineColumnContent from '@opentripplanner/itinerary-body/lib/otp-react-redux/line-column-content'
+import PlaceName from '@opentripplanner/itinerary-body/lib/otp-react-redux/place-name'
+import RouteDescription from '@opentripplanner/itinerary-body/lib/otp-react-redux/route-description'
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+
+import { showLegDiagram } from '../../../actions/map'
+import { setActiveLeg } from '../../../actions/narrative'
+import { setViewedTrip } from '../../../actions/ui'
+import TransitLegSubheader from './connected-transit-leg-subheader'
+import TripDetails from '../connected-trip-details'
+import TripTools from '../trip-tools'
+
+const noop = () => {}
+
+class ConnectedItineraryBody extends Component {
+  shouldComponentUpdate (nextProps, nextState) {
```

Comment:
This shouldComponentUpdate deserves a comment about the conditions/user interactions it is trying to account for.